cmake/gen_version_h.cmake

```diff
@@ -0,0 +1,10 @@
+# SPDX-License-Identifier: Apache-2.0
+
+cmake_minimum_required(VERSION 3.20.0)
+
+if(NOT DEFINED BUILD_VERSION)
+  include(${ZEPHYR_BASE}/cmake/git.cmake)
```

Comment:
git.cmake now has an unfortunate name.

It exists only to set BUILD_VERSION, so let's either rename it to not have such a generic name, or inline it's code into this if-statement.

I'd prefer inlining the code.

Reply:
nice proposal.

Reply:
done, `git.cmake` now removed and code is inlined.